gnucash/gnome-utils/dialog-tax-table.c

```diff
@@ -625,19 +620,22 @@ const char
 void
 tax_table_edit_table_cb (GtkButton *button, TaxTableWindow *ttw)
 {
+    const char *oldname;
     const char *newname;
     g_return_if_fail (ttw);
 
     if (!ttw->current_table)
         return;
 
-    newname = char_dialog (NULL, "Rename", "Rename Tax Table", NULL,
-                           gncTaxTableGetName (ttw->current_table));
+    oldname = gncTaxTableGetName (ttw->current_table);
+    newname = char_dialog (ttw->dialog, (_("Rename")),
+                           (_("Rename Tax Table")),
+                           NULL, oldname);
 
-    if (!(newname == NULL || *newname == '\0'))
-      {
+    if (!(newname == NULL || *newname == '\0' || *newname == *oldname))
+    {
```

Comment:
I would change this to...
    if (newname && *newname != '\0')
    {
        if (g_strcmp0 (oldname, newname) != 0)
            gncTaxTableSetName (ttw->current_table, newname);
    }